client/src/App.js

```diff
@@ -10,25 +10,65 @@ import OnBoardingPage from "./pages/OnBoarding/OnBoarding";
 import SignUpPage from "./pages/Signup";
 import Scheduler from "./pages/Scheduler";
 import Authentication from "./pages/Authentication";
+import axios from "axios";
 
 import "./App.css";
 
 function App() {
-    return (
-        <MuiThemeProvider theme={theme}>
-            <BrowserRouter>
-                <Route exact path="/">
-                    <Redirect to="/signup" />
-                </Route>
-                <Route path="/signup" component={SignUpPage} />
-                <Route path="/login" component={LogInPage} />
-                <Route path="/onboarding" component={OnBoardingPage} />
-                <Route path="/authorized" component={Authentication} />
-                <Route path="/schedule-meeting" component={Scheduler} />
-                <Route path="/home" component={Home} />
-            </BrowserRouter>
-        </MuiThemeProvider>
-    );
+  const [authenticated, setAuthenticated] = useState(false);
+
+  useEffect(async () => {
+    try {
+      const resp = await axios.get(
+        "http://localhost:3001/api/authentication/test",
+        { withCredentials: true }
+      );
+      if (resp.status === 200) {
+        setAuthenticated(true);
+      }
+    } catch (err) {
+      setAuthenticated(false);
+    }
+  }, []);
+
+  return (
+    <MuiThemeProvider theme={theme}>
+      <BrowserRouter>
+        <Route exact path="/">
+          <Redirect to="/signup" />
+        </Route>
+        <Route
+          path="/signup"
+          render={() =>
+            authenticated ? <Redirect to="/home" /> : <SignUpPage />
+          }
+        />
```

Comment:
we can abstract this out to a "PrivateRoute" component. It will take the `SignUpPage` component as a prop, do the auth check, and display the component if authenticated.
[This is the general idea](https://medium.com/@tomlarge/private-routes-with-react-router-dom-28e9f40c7146)